oauthproxy.go

```diff
@@ -239,6 +243,11 @@ func NewOAuthProxy(opts *Options, validator func(string) bool) *OAuthProxy {
 		refresh = fmt.Sprintf("after %s", opts.CookieRefresh)
 	}
 
+	whitelistDomains := opts.WhitelistDomains
+	if strings.HasPrefix(opts.CookieDomain, ".") {
```

Comment:
I don't think this is necessarily what we want to do with the whitelist domains, automatically whitelisting the entire list of subdomains was not the intention for this flag. You may want to allow the cookie to be valid for multiple domains, yet the proxy to only actually authenticate against 2 or 3 subdomains. By doing this change, we remove that behaviour if I've understood correctly

Reply:
Updated the logic to only do this when `whitelistDomains` is not configured and subdomain routing is used.

It is necessary because when subdomain routing enabled, the redirect URL is always a full URL, which always fails unless `whitelistDomains` is configured.